### PR TITLE
Article header

### PR DESCRIPTION
**Pull request by walsh9**
- Add article page header
- Add top banner with ad slot
- Refactor sidebar to be independent from main header, so Article header can use the same component instance
- Refactor donateUrlBase and utmCampaign parameters to be consistent across components

### CODE REVIEW

File: components/ArticlePageHeader.vue [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+  position: fixed;
+  z-index: 1000;
+  top: 116px;
+  background: white;
+  @include media('>md') {
+    top: 0px;
+  }
+}
+.article-page-header-progress {
+  --progressPercentage: v-bind(progressPercentage);
```
Comment (kimlarocca) [inline components/ArticlePageHeader.vue:152]: Nice! Forgot about this feature

**Review by kimlarocca — APPROVED** [review]

File: components/ArticlePageHeader.vue [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+    <div class="article-page-header-contents p-3 md:px-5 flex justify-content-between">
+      <div class="article-page-header-left">
+        <v-flexible-link to="/" raw>
+          <LogoGothamist class="article-page-header-logo" />
+        </v-flexible-link>
+        <div class="article-page-header-tagline hidden">News for New Yorkers</div>
+        <div class="hidden">{{props.title}}</div>
+      </div>
+      <div class="article-page-header-center hidden md:block px-4">
+        <span class="article-page-header-title">{{title}}</span>
```
Comment (bonomite) [inline components/ArticlePageHeader.vue:50]: <h6>{{ title }}</h6>

File: components/ArticlePageHeader.vue [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+  })
+  const sidebarIsOpen = useSidebarIsOpen()
+  const progressPercentage = computed(() => `${props.progress}%`)
+  const openSidebar = () => { sidebarIsOpen.value = true }
+</script>
+
+<template>
+<header class="article-page-header">
+  <div class="article-page-header-progress">
+    <div class="article-page-header-contents p-3 md:px-5 flex justify-content-between">
```
Comment (bonomite) [inline components/ArticlePageHeader.vue:41]: add "align-items-center" to the class list

File: components/ArticlePageHeader.vue (hunk 2 above)
Comment (bonomite) [reply components/ArticlePageHeader.vue:50]: you can remove the ".article-page-header-title" in the scss

Comment (bonomite) [inline components/ArticlePageHeader.vue:50]: - add `h6` class to the parent div 
`<div class="article-page-header-center h6 hidden md:block px-4">`

- remove <span> and just leave the `{{title}}`

File: components/ArticlePageHeader.vue [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+.article-page-header-right {
+  min-width: 135px;
+  @include media('>lg') {
+    min-width: 280px;
+  }
+}
+
+.article-page-header-logo {
+  width: auto;
+  height: 36px;
```
Comment (bonomite) [inline components/ArticlePageHeader.vue:193]: height: 34px;

File: components/ArticlePageHeader.vue [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+              })
+            "
+          />
+        </v-share-tools>
+        <v-flexible-link :to="`${donateUrlBase}&utm_campaign={utmCampaign}`" raw class="pr-2">
+            <Button class="article-page-header-donate-button p-button-rounded"> 
+                <span class="p-button-label">Donate</span>
+            </Button>
+        </v-flexible-link>
+        <Button icon="pi pi-bars" class="p-button p-component p-button-icon-only p-button-text p-button-rounded" @click="openSidebar" />
```
Comment (bonomite) [inline components/ArticlePageHeader.vue:132]: add `-mr-2` to the class list

File: components/ArticlePageHeader.vue (hunk 3 above)
Comment (bonomite) [reply components/ArticlePageHeader.vue:41]: replace `p-3` with `py-3 px-4`

File: components/ArticlePageHeader.vue (hunk 5 above)
Comment (bonomite) [reply components/ArticlePageHeader.vue:132]: also do this to the hamburger in the GothamistMainHeader

Comment (bonomite) [reply components/ArticlePageHeader.vue:132]: also add "align-items-center" here in the GothamistMainHeader file
image

Comment (bonomite) [reply components/ArticlePageHeader.vue:132]: this final adjustment will make the headers identical

File: pages/[sectionSlug]/[articleSlug].vue [hunk 6]
```diff
@@ -22,11 +23,17 @@ if (article.leadGallery) {
...
 const galleryLength = gallery?.slides.length || 0
-
+const config = useRuntimeConfig()
 const trackingData = useArticlePageTrackingData(article)
 const adTargetingData = useArticlePageAdTargetingData(article)
 const sensitiveContent = useSensitiveContent()
 const headMetadata = useArticlePageHeadMetadata(article)
+const { y: scrollY } = useWindowScroll()
+const isMediumOrUpScreen = useMediaQuery(`(min-width: 768px)`)
+// maybe these values should be calculated automatically by measuring on the position of the main header
```
Comment (bonomite) [inline pages/[sectionSlug]/[articleSlug].vue:33]: we can use gsap scrollTrigger in the future to determine when the regular header leaves the viewport.

File: components/ArticlePageHeader.vue (hunk 2 above)
Comment (walsh9) [reply components/ArticlePageHeader.vue:50]: actually should this be `props.title`?

Comment (bonomite) [reply components/ArticlePageHeader.vue:50]: I don't think you have to specify if you don't have a var conflict... but I always put props.var for clarity

Comment (walsh9) [reply components/ArticlePageHeader.vue:50]: ah right, you don't need props in the template, just in scripts

**Review by bonomite — APPROVED** [review]
nice

**Pull request merged**